### PR TITLE
Add auto panel fallback

### PR DESCRIPTION
**Pull request by akhilles**
> [!NOTE]
> **Medium Risk**
> Changes fabrication panel geometry and validation for edge-case boards; behavior is test-covered but affects manufacturing output dimensions.
> 
> **Overview**
> **Auto board array create** no longer fails when the default A-series auto plan cannot fit the board. If `auto_board_array_plan` errors and no sheet was requested, it now builds a **1×1** panel with computed courtyard margins and minimum edge rails, using a new **`AutoMinimumPanel`** validation mode.
> 
> That mode **skips the usual 70–297 mm array size limits** (only requires positive width/height) and treats auto margins like other auto paths (non-negative board margins, edge rails ≥ minimum). **Explicit sheet** selection (`auto_board_array_plan_for_sheet`) is unchanged and still errors when the board cannot fit.
> 
> Tests cover large courtyard-driven margins, the oversized-board fallback panel, and that a requested A4 sheet still fails when appropriate.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f46bdcc12790cb36d5eb003d0dccaa98d79985c0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by LK — APPROVED** [review]

File: crates/pcb-ipc2581-tools/src/commands/board_array.rs [hunk 1]
```diff
@@ -416,34 +417,56 @@ fn create_auto_board_array_xml_with_sheet(
...
+                sheet,
+            )?),
+            BoardArrayValidationMode::Auto,
+        )),
+        None => match auto_board_array_plan(board_width, board_height, board_margin) {
+            Ok(plan) => Ok((options_for_auto_plan(plan), BoardArrayValidationMode::Auto)),
+            Err(_) => Ok((
+                minimum_single_board_auto_options(board_margin),
+                BoardArrayValidationMode::AutoMinimumPanel,
+            )),
```
[the remark this answers is not in the document]
Comment (akhilles) [reply crates/pcb-ipc2581-tools/src/commands/board_array.rs:449]: intentional
